doc/contributing.md

```diff
@@ -0,0 +1,247 @@
+
+### Branches
+Branches are like pieces of scrap paper. You can use them to organize your work 
+and solutions to the test questions. You should not have work regarding 
+different problems mixed across several pieces of paper; you may get your 
+progress confused. You should use one (or several) peices of paper for each
+problem you are trying to solve, but should never use one sheet for multiple
+questions.
+
+For RoboCup you should create a new branch for every new item you'd like to work
+on and for every bug or issue you have to fix. This ensures your master branch
+stays clean. You should never solve more than one issue at a time and you should
+never have changes or additions for multiple things in the same branch. You can 
+look at a typical branching layout ![branchingModel](http://justinhileman.info/article/changing-history/git-flow.png).
+
+### Remotes
+If Git is like a test, then remotes would be cheating. A remote allows you to
+view the solution(s) of another classmate, and pull those additions into your 
+repostory as if they were on the classroom whiteboard.
+
+In software this can be particularly useful if a team member is working on some
+new code that may not be perfect yet, but isn't ready to be folded into master. 
+This may happen when you cannot continue work without getting the somewhat 
+related progress from someone else. You should understand that when pulling from 
+someone else, they take no responsibility for any problems their updates may
+cause for you. This is a decently advanced concept for those new to distributed 
+environments, and won't be used too often. We encourage you learn more about 
+this independently if interested. 
+
+### Overall Workflow (important)
+If you read the previous sections, you may be a little overwhelmed. This section
+will describe how these elements interact to form a coherent workflow that will 
+allow you to make contributions more easily. You can view an overall diagram of 
+how data moves between team members and GitHub ![githubDataFlow](http://www.dalescott.net/wp-content/uploads/2012/09/centralized-github-4.png).
+
+Ensure you have a fork of the main repository and that you've cloned it onto
+your desktop. 
+
+You now have a copy of your repository's master branch avaliable to you. When
+you have an idea of what you'd like to contribute, create a new branch before
+starting work. Assume you want to write some radio firmware, so you name your
+branch radioFirmware.
+
+Your new branch contains a copy of the content of master. Make your additions 
+and edits now, they will only affect the radioFirmware branch. When done add 
+and commit the files.
+
+You now have a branch with your contribution, but you haven't contributed until
+the code makes it into the main repository. This involves several steps. First,
+any changes others have made in the team's repository need to be merged into 
+your code. If there are any conflics Git can't resolve automatically, it is 
```

Comment:
It might be good to link to a guide on resolving merge conflicts here.  GitHub has a good one, but there are dozens of others too.

https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/